modules/finance/src/main/java/com/opengamma/strata/finance/rate/bond/FixedCouponBond.java

```diff
@@ -0,0 +1,784 @@
+import com.opengamma.strata.basics.currency.CurrencyAmount;
+import com.opengamma.strata.basics.currency.Payment;
+import com.opengamma.strata.basics.date.DayCount;
+import com.opengamma.strata.basics.date.DaysAdjustment;
+import com.opengamma.strata.basics.schedule.PeriodicSchedule;
+import com.opengamma.strata.basics.schedule.Schedule;
+import com.opengamma.strata.basics.schedule.SchedulePeriod;
+import com.opengamma.strata.collect.id.StandardId;
+
+/**
+ * A fixed coupon bond.
+ * <p>
+ * A fixed coupon bond is a financial instrument that represents a stream of fixed payments. 
+ * The payments consist two types: periodic coupon payments and nominal payment.
+ * The periodic payments are made {@code n} times a year with a fixed coupon rate at individual coupon dates.   
+ * The nominal payment is the unique payment at the final coupon date.
+ * <p>
+ * The periodic coupon payment schedule is defined using {@link PeriodicSchedule}. 
+ * The payment amount is computed with {@code fixedRate} and {@code notionalAmount}. 
+ * The nominal payment is defined from the last period of the periodic coupon payment schedule and {@code notionalAmount}. 
+ * <p>
+ * The accrual factor between two dates is computed {@code dayCount}. 
+ * The legal entity of this fixed coupon bond is identified by {@link StandardId}. The enum, {@link YieldConvention}, 
+ * specifies the yield computation convention, and {@link DaysAdjustment} does the number of days between valuation 
+ * date and settlement date. 
+ * </ul>
+ */
+@BeanDefinition
+public final class FixedCouponBond
+    implements FixedCouponBondProduct, ImmutableBean, Serializable {
+
+  /**
+   * The notional amount of the product.
+   * <p>
+   * The amount is signed.
+   * A positive amount indicates the payments are to be received.
+   * A negative amount indicates the payments are to be paid.
+   * <p>
+   * This must be specified in one of the two currencies of the forward.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final CurrencyAmount notionalAmount;
+  /**
+   * The accrual schedule.
+   * <p>
+   * This is used to define the accrual periods.
+   * These are used directly or indirectly to determine other dates in the product.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final PeriodicSchedule periodicSchedule;
+  /**
+   * The fixed coupon rate. 
+   * <p>
+   * The periodic payments are based on this fixed coupon rate.
+   */
+  @PropertyDefinition
+  private final double fixedRate;
+  /**
+   * The day count convention applicable. 
+   * <p>
+   * The conversion from dates to a numerical value is made based on this day count. 
+   * For the fixed bond, the day count convention is used to compute accrued interest.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final DayCount dayCount;
+  /**
+   * Yield convention.
+   * <p>
+   * The convention defines accrued interest calculation type of the product.  
+   */
```

Comment:
replace by something like: The convention defines how to convert from yield to price and inversely."